File: w2/ps/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h> // get_string(); strlen()
#include <ctype.h> // toupper()
#include <math.h> // round

int main(void)
{
    float letters = 0;
    float words = 1; // words == spaces + 1
    float sentences = 0;

// Didn't implement with functions. Faster to just loop through the text once
    string text = get_string("Text: ");
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i])) // character is a letter
        {
            letters++;
        }
        else if (text[i] == 32) // character is a space
        {
            words++;
        }
        else if (text[i] == 33 || text[i] == 46 || text[i] == 63) // '!', '.', '?'
        {
            sentences++;
        }
    }

    float L = letters / words * 100;
    float S = sentences / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int X = (int) round(index);

    if (X > 15)
    {
        printf("Grade 16+\n");
    }
    else if (X < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", X);
    }
}

File: w6/sentimental-readability/readability.py
# Determines reading level of input text

from cs50 import get_string


def main():
    text = get_string('Text: ')

    l = 0
    s = 0
    w = 1  # words = num of spaces + 1, also no divide by 0 error in C-L func
    for char in text:  # increment different counts
        if char in ['.', '!', '?']:
            s += 1
        elif char.isalpha():
            l += 1
        elif char == ' ':
            w += 1

    grade = round(coleman_liau(l, s, w))

    if grade < 1:
        print('Before Grade 1')
    elif grade >= 16:
        print('Grade 16+')
    else:  # all special cases handled
        print('Grade', grade)


def coleman_liau(letters, sentences, words):  # not necessary, just nice to be modular
    l = letters / words * 100
    s = sentences / words * 100
    return 0.0588 * l - 0.296 * s - 15.8


main()